dockerfiles/Dockerfile-rosetta

```diff
@@ -1,50 +1,316 @@
+# buster-slim configures apt to not store any cache, so no need to rm it
+# TODO: make sure this is the minimum runtime deps
+RUN apt-get -y update \
+  && apt -y install \
+    curl \
+    ca-certificates \
+    dnsutils \
     dumb-init \
-    libssl1.1 \
-    libprocps6 \
+    libffi6 \
     libgmp10 \
     libgomp1 \
-    libffi6 \
-    apt-transport-https \
-    ca-certificates \
-    dnsutils \
-    tzdata \
+    libjemalloc-dev \
+    libssl1.1 \
     postgresql \
-    postgresql-contrib && \
-  rm -rf /var/lib/apt/lists/*
-
-# coda and archive packages
-RUN echo "deb [trusted=yes] http://packages.o1test.net $deb_repo main" > /etc/apt/sources.list.d/coda.list \
-  && apt-get update \
-  && apt-get install --force-yes coda-testnet-postake-medium-curves=0.0.12-beta+403845-develop-0116d7c-PV947be79c -y \
-  && apt-get install --force-yes coda-archive -y
-
-RUN mkdir -p /data/coda-config \
-    && mkdir -p /data/postgresql \
-    && chown postgres /data/postgresql \
-    && su postgres -c '/usr/lib/postgresql/10/bin/initdb -D /data/postgresql' \
-    && echo "data_directory = '/data/postgresql'" >> /etc/postgresql/10/main/postgresql.conf \
-    && echo "listen_addresses='*'" >> /etc/postgresql/10/main/postgresql.conf \
-    && echo "host all  all    0.0.0.0/0  md5" >> /etc/postgresql/10/main/pg_hba.conf
+    postgresql-contrib \
+    tzdata
 
-USER postgres
 
-COPY src/app/archive/create_schema.sql .
+# don't use the released package for the rosetta image as we just built it in docker
+# we could use a deb package pulled from the builder though, especially if the deb includes other useful items for
+# rosetta. Currently we're not building the deb here as it requires a tools that we /dont/ need in rosetta.
+#RUN echo "deb [trusted=yes] http://packages.o1test.net $deb_repo main" > /etc/apt/sources.list.d/coda.list \
+#  && apt-get update \
+#  && apt-get install --force-yes coda-testnet-postake-medium-curves=0.0.12-beta+403845-develop-0116d7c-PV947be79c -y \
+#  && apt-get install --force-yes coda-archive -y
+
+# Make all directories for configuration and empty s3 cache dirs
+RUN mkdir -p /tmp/s3_cache_dir \
+    &&  mkdir -p /tmp/coda_cache_dir \
+    && mkdir -p ${POSTGRES_DATA_DIR} \
+    && chown postgres ${POSTGRES_DATA_DIR}
+
+COPY --from=builder /home/opam/app/* /coda-bin/
+COPY --from=builder /home/opam/coda/src/app /app
+COPY --from=builder /home/opam/coda/src/app/rosetta/config.json /data/config.json
 
-RUN /etc/init.d/postgresql start \
+# Initial image could be made smaller at the expense of start up time by removing these
+# because all s3-cached files will be downloaded as needed at runtime
+
+# Remove these to optimize image size, start up time is already bad because testnets each need their own genesis proof
+COPY --from=builder /tmp/s3_cache_dir /tmp/s3_cache_dir
+COPY --from=builder /tmp/coda_cache_dir /tmp/coda_cache_dir
+
+USER postgres
+
+# TODO: use POSTGRES_DATA_DIR variable more consistently and ensure proper escaping
```

Comment:
Worth storing all of this logic in a repo script so that it can be iterated on independently of the Dockerfile and also help cut down on some of the logic in the Dockerfile?

Reply:
Noted as a todo, centralizing this would be really nice, there are already a variety of scripts for this sort of purpose, but it would need to be generalized to support each use case. The script should probably be maintained alongside the archive node.